src/main/kotlin/interactor/GetCityHasHighestRentingBetweenCenterAndOutsideCityCenter.kt

```diff
@@ -0,0 +1,38 @@
+package interactor
+import model.CityEntity
+import model.RealEstatesPrices
+
+class GetHighestApartmentPriceDifferenceCityInteractor(
+    private val dataSource: CostOfLivingDataSource
+) {
+
+    fun execute(cities: List<CityEntity>): CityEntity? {
+        return dataSource
+            .getAllCitiesData()
+            .filter(::excludeNullPricesAndLowQualityData)
+            .maxByOrNull { it.realEstatesPrices.getApartmentPriceDifference() }
+    }
+
+
+    private fun excludeNullPricesAndLowQualityData(city: CityEntity): Boolean {
+        with(city.realEstatesPrices) {
+            return apartmentOneBedroomInCityCentre != null &&
+                    apartmentOneBedroomOutsideOfCentre != null &&
+                    apartment3BedroomsInCityCentre != null &&
+                    apartment3BedroomsOutsideOfCentre != null &&
+                    pricePerSquareMeterToBuyApartmentInCityCentre != null &&
+                    pricePerSquareMeterToBuyApartmentOutsideOfCentre != null &&
```

Comment:
pricePerSquareMeterToBuyApartmentInCityCentre &&   pricePerSquareMeterToBuyApartmentOutsideOfCentre
remove it, because your task need rent apartment not Buy Apartment.